corehq/motech/dhis2/entities_helpers.py

```diff
@@ -287,31 +305,50 @@ def create_relationships(
                 "case type."
             ).format(subcase_type=subcase_trigger_info.type,
                      supercase_type=supercase.type))
-        supercase_info = get_case_trigger_info_for_case(
+        supercase_trigger_info = get_case_trigger_info_for_case(
             supercase,
             [supercase_config.tei_id],
         )
-        supercase_tei_id = get_value(supercase_config.tei_id, supercase_info)
-        if not supercase_tei_id:
+        supercase_tracked_entity_instance_id = get_value(supercase_config.tei_id, supercase_trigger_info)
+        if not supercase_tracked_entity_instance_id:
             # Registering the supercase in DHIS2 failed.
             raise ConfigurationError(_(
                 'Unable to create DHIS2 relationship: The case {case} is not '
                 'registered in DHIS2.'
-            ).format(case=supercase_info))
-        if rel_config.supercase_to_subcase_dhis2_id:
-            create_relationship(
-                requests,
-                rel_config.supercase_to_subcase_dhis2_id,
-                supercase_tei_id,
-                subcase_tei_id,
-            )
-        if rel_config.subcase_to_supercase_dhis2_id:
-            create_relationship(
-                requests,
-                rel_config.subcase_to_supercase_dhis2_id,
-                subcase_tei_id,
-                supercase_tei_id,
-            )
+            ).format(case=supercase_trigger_info))
+        supercase_to_subcase_relationship_spec = RelationshipSpec(
+            relationship_type_id=relationship_config.supercase_to_subcase_dhis2_id,
+            from_tracked_entity_instance_id=supercase_tracked_entity_instance_id,
+            to_tracked_entity_instance_id=subcase_tracked_entity_instance_id
+        )
+        subcase_to_supercase_relationship_spec = RelationshipSpec(
+            relationship_type_id=relationship_config.subcase_to_supercase_dhis2_id,
+            from_tracked_entity_instance_id=subcase_tracked_entity_instance_id,
+            to_tracked_entity_instance_id=supercase_tracked_entity_instance_id,
+        )
+        ensure_relationship_exists(
+            requests=requests,
+            relationship_spec=supercase_to_subcase_relationship_spec,
+            existing_relationships=tracked_entity_relationships
+        )
+        ensure_relationship_exists(
+            requests=requests,
+            relationship_spec=subcase_to_supercase_relationship_spec,
+            existing_relationships=tracked_entity_relationships
+        )
+
+
+def ensure_relationship_exists(requests, relationship_spec, existing_relationships):
+    """Checks to see if `relationship_spec` is in `existing_relationships`. If not, we try to create the
+    relationship represented by `relationship_spec`.
+    """
+    for existing_spec in existing_relationships:
+        relationships_match = relationship_spec.as_dict() == existing_spec.as_dict()
```

Comment:
Very nit, but this can be shortened...

```suggestion
        if relationship_spec.as_dict() == existing_spec.as_dict():
            return None
```